btsieve/src/bitcoin/bitcoind_http_blocksource.rs

```diff
@@ -0,0 +1,205 @@
+use crate::blocksource::{self, BlockSource};
+use bitcoin_support::{deserialize, Block, Network, Transaction};
+use futures::{Future, Stream};
+use reqwest::r#async::Client;
+use serde::Deserialize;
+use std::time::Duration;
+use tokio::timer::Interval;
+
+#[derive(Deserialize)]
+struct ChainInfo {
+    bestblockhash: String,
+}
+
+#[derive(Debug)]
+pub enum Error {
+    Reqwest(reqwest::Error),
+    Hex(hex::FromHexError),
+    BlockDeserialization(bitcoin_support::consensus::encode::Error),
+    TransactionDeserialization(String),
+}
+
+#[derive(Clone)]
+pub struct BitcoindHttpBlockSource {
+    network: Network,
+    base_url: String,
```

Comment:
```suggestion
    base_url: Url,
```
This would allow some free validation.